api/provider/aws/service_delete.go

```diff
@@ -1,5 +1,130 @@
+func (s *ServiceProvider) Delete(serviceID string) error {
+	desiredCount := 0
+
+	environmentID, err := lookupEntityEnvironmentID(s.TagStore, "service", serviceID)
+	if err != nil {
+		if err, ok := err.(*errors.ServerError); ok && err.Code == errors.ServiceDoesNotExist {
+			return nil
+		}
+
+		return err
+	}
+
+	clusterName := addLayer0Prefix(s.Config.Instance(), environmentID)
+	fqServiceID := addLayer0Prefix(s.Config.Instance(), serviceID)
+
+	service, err := s.readService(clusterName, fqServiceID)
+	if err != nil {
+		return err
+	}
+
+	taskARNs, err := s.getServieTasks(clusterName, service.Deployments)
+	if err != nil {
+		return err
+	}
+
+	if err := s.stopServiceTasks(clusterName, fqServiceID, taskARNs); err != nil {
+		return err
+	}
+
+	if err := s.scaleService(clusterName, fqServiceID, desiredCount); err != nil {
+		return err
+	}
+
+	if err := s.deleteService(clusterName, fqServiceID); err != nil {
+		return err
+	}
+
+	if err := deleteEntityTags(s.TagStore, "service", serviceID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) getServieTasks(clusterName string, deployments []*ecs.Deployment) ([]string, error) {
+	taskARNs := []string{}
+
+	for _, deployment := range deployments {
+		clusterTaskARNsStopped, err := listClusterTaskARNs(s.AWS.ECS, clusterName, aws.StringValue(deployment.Id), ecs.DesiredStatusStopped)
+		if err != nil {
+			return nil, err
+		}
+
+		clusterTaskARNsRunning, err := listClusterTaskARNs(s.AWS.ECS, clusterName, aws.StringValue(deployment.Id), ecs.DesiredStatusRunning)
+		if err != nil {
+			return nil, err
+		}
+
+		taskARNs = append(taskARNs, clusterTaskARNsStopped...)
+		taskARNs = append(taskARNs, clusterTaskARNsRunning...)
+	}
+
+	return taskARNs, nil
+}
+
+func (s *ServiceProvider) stopServiceTasks(clusterName, serviceID string, taskARNs []string) error {
+	for _, taskARN := range taskARNs {
+		input := &ecs.StopTaskInput{}
+		input.SetCluster(clusterName)
+		input.SetTask(aws.StringValue(&taskARN))
+
+		if err := input.Validate(); err != nil {
+			return err
+		}
+
+		if _, err := s.AWS.ECS.StopTask(input); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) scaleService(clusterName, serviceID string, desiredCount int) error {
+	input := &ecs.UpdateServiceInput{}
+
+	input.SetDesiredCount(int64(desiredCount))
+	input.SetCluster(clusterName)
+	input.SetService(serviceID)
+
+	if err := input.Validate(); err != nil {
+		return err
+	}
+
+	if _, err := s.AWS.ECS.UpdateService(input); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *ServiceProvider) deleteService(clusterName, serviceID string) error {
+	input := &ecs.DeleteServiceInput{}
+	input.SetCluster(clusterName)
+	input.SetService(serviceID)
+
+	if err := input.Validate(); err != nil {
+		return err
+	}
+
+	if _, err := s.AWS.ECS.DeleteService(input); err != nil {
+		if err, ok := err.(awserr.Error); ok && err.Code() == errors.ServiceDoesNotExist.String() {
```

Comment:
This is mixing the `aws-sdk-go` error codes and our error codes in `layer0/common/errors`. They are two very different things and shouldn't be mixed w/ each other. Any overlap between the two is purely coincidental. 

I don't know the exact error code returned by ECS, but this should look something like:
```
if err, ok := err.(awserr.Error); ok && err.Code() == "SomeECSErrorCode" {
```